src/Core/Infrastructure/Service/Filesystem.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace Core\Infrastructure\Service;
+
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem as SymfonyFilesystem;
+
+class Filesystem extends SymfonyFilesystem
+{
+    /**
+     * @var string|null
+     */
+    protected static ?string $lastError;
+
+    /**
+     * @param string $filename
+     * @param bool $useIncludePath
+     * @param mixed $context
+     *
+     * @throws \Symfony\Component\Filesystem\Exception\IOException
+     *
+     * @return string
+     */
+    public function readFile(
+        string $filename,
+        bool $useIncludePath = false,
+        $context = null
+    ): string {
+        $data = $context === null
+            ? static::box('file_get_contents', $filename, $useIncludePath)
+            : static::box('file_get_contents', $filename, $useIncludePath, $context);
+
+        if ($data === false) {
+            throw new IOException(sprintf('Failed to read "%s" file: %s', $filename, static::$lastError));
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param string $directory
+     * @param int $sorting_order
+     * @param mixed $context
+     *
+     * @throws \Symfony\Component\Filesystem\Exception\IOException
+     *
+     * @return array<string>
+     */
+    public function scanDir(string $directory, int $sorting_order = SCANDIR_SORT_ASCENDING, $context = null): array
+    {
+        $data = $context === null
+            ? static::box('scandir', $directory, $sorting_order)
+            : static::box('scandir', $directory, $sorting_order, $context);
```

Comment:
You don't need to check here if context is null or not, it is null by default in scandir(). So it doesn't change anything if you will always call: 

```
$data = static::box('scandir', $directory, $sorting_order, $context);
```

The same is valid for file_get_contents()